src/repository.js

```diff
@@ -39,30 +40,78 @@ module.exports = class Repository {
     return entities
   }
 
-  async findBy(search) {
+  /** 
+  *
+  * Find all method
+  * 
+  * @param {type}   object.limit Limit items to list  
+  * @param {type}   object.orderBy Order by query
+  * @param {type}   object.offset Rows that will be skipped from the resultset
+  *
+  * @return {type} List of entities
+  */
+   async findAll(options = {
+    limit: 0,
+    offset: 0,
+    orderBy: null
+  }) {
+
+    const entities = this.find({ limit: options.limit, offset: options.offset, orderBy: options.orderBy })
+    return entities
+  }
+  
+  /** 
+  *
+  * Find entities
+  * 
+  * @param {type}   object.limit Limit items to list  
+  * @param {type}   object.offset Rows that will be skipped from the resultset
+  * @param {type}   object.search Where query term
+  * @param {type}   object.orderBy Order by query
+  *
+  * @return {type} List of entities
+  */
+  async find(options = {
+    limit: 0,
+    offset: 0,
+    orderBy: null,
+    conditions: null
+  }) {
+
+    options.orderBy = options.orderBy || null
+    options.limit = options.limit || 0
+    options.offset = options.offset || 0
+    options.conditions = options.conditions || null
 
     const tableFields = this.dataMapper.tableFields()
-    const searchTermTableField = this.dataMapper.toTableFieldName(Object.keys(search)[0])
-    const searchTerm = Object.keys(search)[0]
-    if (!searchTerm || searchTerm === "0") throw "search term is invalid"
-
-    const searchValue = Array.isArray(search[searchTerm])
-      ? search[searchTerm]
-      : [search[searchTerm]]
-
-    if (
-      !search[searchTerm] ||
-      (typeof search[searchTerm] === "object" &&
-        !Array.isArray(search[searchTerm])) ||
-      (Array.isArray(search[searchTerm]) && !search[searchTerm].length)
-    )
-      throw "search value is invalid"
 
-    const ret = await this.runner
+    let query = this.runner
       .select(tableFields)
-      .whereIn(searchTermTableField, searchValue)
+
+    if (options.limit > 0) query = query.limit(options.limit)
+    if (options.offset > 0) query = query.offset(options.offset)
+
+    if (options.conditions) {
+      const searchTermTableField = this.dataMapper.toTableFieldName(Object.keys(options.conditions)[0])
```

Comment:
`searchTermTableField`, `searchTerm` and `searchValue` var names should reflect the change from `search` to `conditions`

Reply:
Done!